File: Mean of range in array.cpp
class Solution {
  public:
    vector<int> findMean(vector<int> &arr, vector<vector<int>> &queries) {
        // code here
        int n = arr.size();
        vector<int> result;
        result.reserve(queries.size());
        
        vector<long long> prefix(n + 1, 0);
        for (int i = 0; i < n; i++) {
            prefix[i + 1] = prefix[i] + arr[i];
        }
        
        for (auto &q : queries) {
            int l = q[0];
            int r = q[1];
            
            long long rangeSum = prefix[r + 1] - prefix[l];
            int length = r - l + 1;
            int mean = rangeSum / length; 
            result.push_back(mean);
        }
        
        return result;
    }
};

File: Mean of range in array.py
class Solution:
    def findMean(self, arr, queries):
        # code here
        n = len(arr)
        
        prefix = [0] * (n + 1)
        for i in range(n):
            prefix[i + 1] = prefix[i] + arr[i]
        
        result = []
        for l, r in queries:
            range_sum = prefix[r + 1] - prefix[l]
            length = r - l + 1
            mean = range_sum // length
            result.append(mean)
            
        return result
